go/cmd/dolt/commands/merge.go

```diff
@@ -171,13 +171,24 @@ func mergeBranch(ctx context.Context, dEnv *env.DoltEnv, dref ref.DoltRef) errha
 	cli.Println("Updating", h1.String()+".."+h2.String())
 
 	if ok, err := cm1.CanFastForwardTo(ctx, cm2); ok {
-		return executeFFMerge(ctx, dEnv, cm2)
+		verr = executeFFMerge(ctx, dEnv, cm2)
```

Comment:
No reason not to immediately return the result of these two functions as in the previous version of this file, then delete the verr != nil block below.

Reply:
Changed it back to returning these methods. Removed verr != nil block. And moved SaveTrackedDocsFromWorking into those functions, next to where the roots are being updated: https://github.com/liquidata-inc/dolt/pull/299/files#diff-1647ea59f1bf8a37af76a621b0ea152fR219